.github/workflows/compatibility-tests.yaml

```diff
@@ -0,0 +1,802 @@
+              exit 1
+            fi
+            sleep 1
+          done
+
+          # validate that helm charts installed using the native helm workflow were deployed via the helm CLI correctly
+
+          if ! helm ls -n postgres-test | awk 'NR>1{print $1}' | grep -q postgresql; then
+            printf "postgresql helm release not found in postgres-test namespace\n\n"
+            helm ls -n postgres-test
+            exit 1
+          fi
+
+          if ! helm ls -n "$APP_SLUG" | awk 'NR>1{print $1}' | grep -q private-chart; then
+            printf "private-chart helm release not found in %s namespace\n\n" "$APP_SLUG"
+            helm ls -n "$APP_SLUG"
+            exit 1
+          fi
+
+      - name: Generate support bundle on failure
+        if: failure()
+        uses: ./.github/actions/generate-support-bundle
+        with:
+          kots-namespace: "$APP_SLUG"
+          aws-access-key-id: '${{ secrets.E2E_SUPPORT_BUNDLE_AWS_ACCESS_KEY_ID }}'
+          aws-secret-access-key: '${{ secrets.E2E_SUPPORT_BUNDLE_AWS_SECRET_ACCESS_KEY }}'
+  
+      - name: Remove Cluster
+        id: remove-cluster
+        uses: replicatedhq/replicated-actions/remove-cluster@v1
+        if: ${{ always() && steps.create-cluster.outputs.cluster-id != '' }}
+        continue-on-error: true
+        with:
+          api-token: ${{ secrets.C11Y_MATRIX_TOKEN }}
+          cluster-id: ${{ steps.create-cluster.outputs.cluster-id }}
+
+
+  validate-kots-pull:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        cluster: ${{ fromJson(inputs.versions-to-test) }}
+    env:
+      APP_NAME: multi-namespace-yeti
+      APP_SLUG: multi-namespace
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Create Cluster 
+        id: create-cluster 
+        uses: replicatedhq/replicated-actions/create-cluster@v1 
+        with:
+          api-token: ${{ secrets.C11Y_MATRIX_TOKEN }}
+          kubernetes-distribution: ${{ matrix.cluster.distribution }}
+          kubernetes-version: ${{ matrix.cluster.version }}
+          cluster-name: automated-kots-${{ github.run_id }}-${{ matrix.cluster.distribution }}-${{ matrix.cluster.version }}
+          timeout-minutes: '120'
+          ttl: 2h
+          instance-type: ${{ matrix.cluster.distribution == 'gke' && 'n2-standard-4' || '' }}
+          export-kubeconfig: true
+
+      - name: download kots binary
+        uses: actions/download-artifact@v3
+        with:
+          name: kots
+          path: bin/
+
+      - run: chmod +x bin/kots
+
+      - name: create namespace and dockerhub secret
+        run: |
+          kubectl create ns "$APP_NAME"
+          kubectl create secret docker-registry kotsadm-dockerhub --docker-server index.docker.io --docker-username "${{ secrets.E2E_DOCKERHUB_USERNAME }}" --docker-password "${{ secrets.E2E_DOCKERHUB_PASSWORD }}" --namespace "$APP_NAME"
```

Comment:
Opted to create the dockerhub secret this way since KOTS needs to be deployed first before `kots docker ensure-secret` can run successfully.  Since we do many headless installs in our CLI tests, images would be checked before the secret can be created in those cases.